components/Map/Map.tsx

```diff
@@ -0,0 +1,221 @@
+            }
+        });
+
+        this.map.on('mouseleave', 'points', () => {
+            this.map.getCanvas().style.cursor = 'auto';
+            if (hoveredStateId) this.map.setFeatureState({
+                source: this.pointsLayerName,
+                id: hoveredStateId,
+            }, {hover: false});
+            hoveredStateId = null;
+        });
+    };
+
+    addPopup = (component, coordinates) => {
+        const placeholder = document.createElement('div');
+        ReactDOM.render(component,
+            placeholder);
+
+        this.popUp = new mapboxgl.Popup({
+            closeButton: true,
+            closeOnClick: true,
+        })
+            .setDOMContent(placeholder)
+            .setLngLat(coordinates)
+            .addTo(this.map);
+    };
+
+    addClick = points => {
+        this.map.on('click', 'points', event => {
+            const id = event.features[0].id;
+            const coordinates = event.features[0].geometry.coordinates.slice();
+            while (Math.abs(event.lngLat.lng - coordinates[0]) > 180) {
+                coordinates[0] += event.lngLat.lng > coordinates[0] ? 360 : -360;
+            }
+
+            this.addPopup(
+                <MapPopUp
+                    onClick={() => this.setState({modalOpen: true})}
+                    id={id}
+                    feature={event.features[0]}/>,
+                coordinates,
+            );
+            this.state.currentHighlight !== null && this.map.setFeatureState({
+                source: this.pointsLayerName,
+                id: this.state.currentHighlight,
+            }, {click: false});
+            this.map.setFeatureState({source: this.pointsLayerName, id}, {click: true});
+            this.setState({
+                currentHighlight: id,
+            });
+        });
+
+        this.map.on('click', event => {
+            // stop propagation of event
+            // mapbox draws on canvas, so event.stopPropagation() in
+            // this.map.on('click', 'points', event => { ... does not
+            // stop the bubblin, so check for if any feature was clicked
+            if (this.map.queryRenderedFeatures(event.point).filter(feature => feature.source === this.pointsLayerName).length === 0) {
+                points.features.forEach(feature => {
+                    this.map.setFeatureState({source: this.pointsLayerName, id: feature.id}, {click: false});
+                });
+                this.setState({currentHighlight: null});
+            }
+        })
+    };
+
+    fitBounds = event => {
+        event.stopPropagation();
+        this.map.fitBounds(this.state.bbox, {padding: 30});
+    };
+
+
+    render() {
+        const {modalOpen} = this.state;
+        return <Fragment>
+            {/* MODAL */}
+            <div
+                style={{
+                    backgroundColor: 'white',
+                    height: '100vh',
+                    width: '100vw',
+                }}
```

Comment:
Die Styles vllt außerhalb der React.Component in einer `const` definieren? So wird ansonsten bei jedem Rerender ein neues Object erstellt.